fastlane_core/lib/fastlane_core/build_watcher.rb

```diff
@@ -0,0 +1,78 @@
+module FastlaneCore
+  class BuildWatcher
+    def self.find_build(app = nil)
+      build = nil
+      app.latest_version.candidate_builds.each do |b|
+        if !build or b.upload_date > build.upload_date
+          build = b
+        end
+      end
+      unless build
+        UI.user_error!("No processing builds available for app #{app.bundle_id}")
+      end
+      return build
+    end
+
+    def self.wait_for_train(app = nil, platform = "ios", sleep_time = 30)
```

Comment:
I am usually defensive when it comes to exposing code public unless needed, so I would prefer it to be private unless there's a strong argument / existing use case.